File: BlackJack.WebApi/Repository/WinningRulesRepository.cs
using BlackJack.WebApi.Interface;
using BlackJack.WebApi.Models;
using Microsoft.Extensions.Configuration;

namespace BlackJack.WebApi.Repository
{
    public class WinningRulesRepository : IWinningRules
    {
        private readonly IConfiguration _configuration;
        private const string _aceAndPontoon = "A";
        private const int _numberOfTurnsPlayed = 5;
        private const int _busts = 21;
        private const int _closeToBlackJack = 20;
        public WinningRulesRepository(IConfiguration configuration)
        {
            _configuration = configuration;
        }

        public bool IsPlayerHasAceAndPontoon(Hand player)
        {
            return player.Name.Equals(_aceAndPontoon) ? false : true;

        }

        public bool IsPlayerHasFiveTurn(Hand player)
        {
            return player.NumberOfTurnsPlayed.Equals(_numberOfTurnsPlayed) && player.Total < _busts ? true : false;
        }

        public WinningRules IsBlackJack(HandCustomModel handCustomModel)
        {
            var businessRules = new WinningRules { };
            if (handCustomModel.Player.Total == _busts)
            {
                businessRules.IsPlayerBlackJack = true;
            }
            if (handCustomModel.Dealer.Total == _busts)
            {
                businessRules.IsTie = true;
                businessRules.IsDealerBlackJack = true;
            }

            return businessRules;
        }

        public WinningRules IsBusts(HandCustomModel handCustomModel )
        {
            var businessRules = new WinningRules { };
            if (handCustomModel.Dealer.Total > _busts && handCustomModel.Player.Total < _busts)
            {
                businessRules.IsDealerBusts = true;
            }

            if (handCustomModel.Dealer.Total < _busts)
            {
                businessRules.IsPlayerBusts = true;
            }
            return businessRules;
        }

        public WinningRules IsCloseToBlackJack(HandCustomModel handCustomModel)
        {
            var businessRules = new WinningRules { };
            if (handCustomModel.Player.Total == _closeToBlackJack && handCustomModel.Player.Total < _busts)
            {
                businessRules.IsPlayerCloseToBlackJack = true;
            }

            if (handCustomModel.Dealer.Total == _closeToBlackJack && handCustomModel.Dealer.Total < _busts)
            {
                businessRules.IsDealerCloseToBlackJack = true;
            }
            return businessRules;
        }
    }
}

File: BlackJack.TypeScript/src/repositories/WinningRulesRepository.ts
import { Hand } from '../models/Hand';
import { HandCustomModel } from '../models/HandCustomModel';
import {WinningRules} from '../models/WinningRules'

 class WinningRulesRepository{

    constructor() { }

      isPlayerHasAceAndPontoon(player :Hand){
          const aceAndPontoon : string ='A';
          const winningRules = new  WinningRules();

          if(player.name == aceAndPontoon )
          {
              winningRules.isPlayerHasAceAndPontoon = true;
          }
          return winningRules;

      }

      isPlayerHasFiveTurn(player:Hand) {
             const numberOfTurns : number = 5;
             const busts : number = 21;
            var winningRules = new WinningRules();
            if (player.numberOfTurnsPlayed == numberOfTurns && player.total < busts)
                winningRules.isPlayerHasFiveTurn = true;
            return winningRules;
        }

        isBlackJack(handCustomModel : HandCustomModel)  {
            const busts : number = 21;
            var businessRules = new WinningRules();
            if (handCustomModel.player.Total == busts)
            {
                businessRules.isPlayerBlackJack = true;
            }
            if (handCustomModel.dealer.Total == busts)
            {
                businessRules.isTie = true;
                businessRules.isDealerBlackJack = true;
            }

            return businessRules;
          }

          isBusts(handCustomModel : HandCustomModel ) {
               const busts : number = 21;
              var businessRules = new WinningRules();
              if (handCustomModel.dealer.Total > busts && handCustomModel.player.Total < busts)
              {
                  businessRules.isDealerBusts = true;
              }
  
              if (handCustomModel.dealer.Total < busts)
              {
                  businessRules.isPlayerBusts = true;
              }
              return businessRules;
          }
  
          isCloseToBlackJack( handCustomModel : HandCustomModel ){
              const closeToBlackJack : number = 20;
              const busts : number = 21;
              var businessRules = new WinningRules();
              if (handCustomModel.player.Total == closeToBlackJack && handCustomModel.player.Total < busts)
              {
                  businessRules.isPlayerCloseToBlackJack = true;
              }
  
              if (handCustomModel.dealer.Total == closeToBlackJack && handCustomModel.dealer.Total < busts)
              {
                  businessRules.isDealerCloseToBlackJack = true;
              }
              return businessRules;
          }

}


export default new WinningRulesRepository();